pkg/tasks/cni.go

```diff
@@ -33,22 +33,19 @@ func ensureCNI(s *state.State) error {
 			return err
 		}
 	case s.Cluster.ClusterNetwork.CNI.WeaveNet != nil:
-		return ensureCNIWeaveNet(s)
+		if s.Cluster.ClusterNetwork.CNI.WeaveNet.Encrypted {
+			if err := weave.EnsureSecret(s); err != nil {
+				return err
+			}
+		}
```

Comment:
What do you think about replacing this with YAML/addon instead? I think it should be possible. We would probably have to add a template function for generating the password.

Reply:
weave.EnsureSecret() will not override secret in case if it's already exists, this is not possible with YAML, and probably there is no point to even try.